util.go

```diff
@@ -17,26 +30,42 @@ func generateSigner() (ssh.Signer, error) {
 }
 
 func parsePtyRequest(s []byte) (pty Pty, ok bool) {
-	term, s, ok := parseString(s)
-	if !ok {
-		return
-	}
-	width32, s, ok := parseUint32(s)
-	if !ok {
-		return
+	reqMsg := &ptyRequestMsg{}
+	err := ssh.Unmarshal(s, reqMsg)
+	if err != nil {
+		return Pty{}, false
 	}
-	height32, _, ok := parseUint32(s)
-	if !ok {
-		return
+
+	modes := []byte(reqMsg.Modelist)
+
+	mode := struct {
+		Key uint8
+		Val uint32
+	}{}
+
+	TerminalModes := make(ssh.TerminalModes, 0)
+	for {
+		if len(modes) < 1 || modes[0] == ttyOPEND || len(modes) < 5 {
```

Comment:
This actually handles 2 different conditions - I'd argue `len(modes) < 1 || modes[0] == ttyOPEND` are success, but `len(modes) < 5` is a failure.